### ISSUE TITLE
Question: Does the support for @JsonViews only work with @RequestBody annotation (io.swagger.v3.oas.annotations.parameters.RequestBody)

### ISSUE THREAD

**Issue by Andrew3431** [springdoc/springdoc-openapi#315]
I want different set of parameters for Request and Response for POST methods. Using @JSONViews #107  seemed to work for displaying different request and response for swagger. However if I use @RequestBody from spring framework (org.springframework.web.bind.annotation.RequestBody) , the JSONView output is not working as expected
The test code mentioned [here](https://github.com/springdoc/springdoc-openapi/tree/master/springdoc-openapi-webmvc-core/src/test/java/test/org/springdoc/api/app54) works well with @RequestBody provided by swagger (io.swagger.v3.oas.annotations.parameters.RequestBody). However trying the swagger @RequestBody , does not read the inputs and provides them with null values.

``` java   
import io.swagger.v3.oas.annotations.parameters.RequestBody;
.
.
@Operation(security = { @SecurityRequirement(name = "api_key")}, summary = "Adds new user.")
@ApiResponses(value = {@ApiResponse(responseCode = "201", description = "the user was created"),
          @ApiResponse(responseCode = "409", description = "the user exists already", content = @Content) })
@RequestMapping(method = RequestMethod.POST}, produces = {APPLICATION_JSON_VALUE})
@ResponseStatus(HttpStatus.CREATED)
@ResponseBody
@JsonView(Views.ReadOnly.class)
    public User create( @Parameter(description = "user information")
            @JsonView(Views.Request.class)
            @RequestBody User user)
             { . .... }
```

where user returns null for variables. Working with @RequestBody from Spring returns the user object with values given during request.  Am I doing something wrong here ?

**Comment by bnasslahsen** [comment]
Hi @Andrew3431, 

 Yes, the support for @JsonViews works only with `@RequestBody` annotation from swagger.
 If you judge its blocking, support for `@RequestBody` from spring, can be added for the next release.

**Comment by Andrew3431** [comment]
Yes, that would be great. The apis don't work well with `@RequestBody` from swagger. Thank you so much.

**Comment by bnasslahsen** [comment]
The support will be available on the next release v1.2.23, of springdoc-openapi.
